src/main/java/com/laserfiche/repository/api/RepositoryApiClientImpl.java

```diff
@@ -25,15 +26,26 @@ public class RepositoryApiClientImpl implements RepositoryApiClient, AutoCloseab
     private final TasksClient tasksClient;
     private final TemplateDefinitionsClient templateDefinitionsClient;
 
-    protected RepositoryApiClientImpl(String servicePrincipalKey, AccessKey accessKey, String baseUrlDebug) {
-        String baseUrl = baseUrlDebug != null ? baseUrlDebug : "https://api." + accessKey.getDomain() + "/repository";
+    protected RepositoryApiClientImpl(String servicePrincipalKey, AccessKey accessKey, String repositoryId,
+            String username,
+            String password, TokenClient client, String baseUrl) {
         httpClient = Unirest.spawnInstance();
-
-        httpClient
-                .config()
-                .setObjectMapper(new RepositoryClientObjectMapper())
-                .interceptor(new OAuthInterceptor(servicePrincipalKey, accessKey));
-
+        if (baseUrl == null) {
+            baseUrl = "https://api." + accessKey.getDomain() + "/repository";
+        } else if (baseUrl.endsWith("/")) {
+            baseUrl.substring(0, baseUrl.length() - 1);
```

Comment:
The `substring` function returns the modified string, I don't think it updates baseUrl. So you still need to assign it.
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#substring(int,%20int)